src/ifaddrs.rs

```diff
@@ -0,0 +1,180 @@
+mod flags {
+    use libc;
+    libc_bitflags!(
+        /// Flags for `InterfaceAddress` entries returned by `getifaddrs`
+        pub flags InterfaceFlags: libc::c_int {
+            IFF_UP,
+            IFF_BROADCAST,
+            IFF_DEBUG,
+            IFF_LOOPBACK,
+            IFF_POINTOPOINT,
+            IFF_NOTRAILERS,
+            IFF_RUNNING,
+            IFF_NOARP,
+            IFF_PROMISC,
+            IFF_ALLMULTI,
+            IFF_MASTER,
+            IFF_SLAVE,
+            IFF_MULTICAST,
+            IFF_PORTSEL,
+            IFF_AUTOMEDIA,
+            IFF_DYNAMIC,
+        }
+    );
+}
+
+#[cfg(not(any(target_os = "linux", target_os = "emscripten", target_os = "fuchsia")))]
+mod flags {
+    use libc;
+    /// Flags for `InterfaceAddress` entries returned by `getifaddrs`
+    libc_bitflags!(
+        pub flags InterfaceFlags: libc::c_int {
+            IFF_LOOPBACK,
+        }
+    );
+}
+
+pub use self::flags::*;
+
+/// Describes a single address for an interface as returned by `getifaddrs`.
+pub struct InterfaceAddress {
+    /// name of the interface
+    pub interface_name: String,
+    /// flags from `SIOCGIFFLAGS`
+    pub flags: InterfaceFlags,
+    /// address of interface
+    pub address: Option<InetAddr>,
+    /// netmask of interface
+    pub netmask: Option<InetAddr>,
+    /// broadcast address of interface, if applicable
+    pub broadcast: Option<InetAddr>,
+    /// point-to-point destination address
+    pub destination: Option<InetAddr>,
+}
+
+impl InterfaceAddress {
+    /// Create an `InterfaceAddress` from the libc struct.
+    fn from_libc_ifaddrs(info: &libc::ifaddrs) -> InterfaceAddress {
+        let ifname = unsafe { ffi::CStr::from_ptr(info.ifa_name) };
+        let address = unsafe { InetAddr::from_libc_sockaddr(info.ifa_addr) };
+        let netmask = unsafe { InetAddr::from_libc_sockaddr(info.ifa_netmask) };
+        let mut addr = InterfaceAddress {
+            interface_name: ifname.to_string_lossy().to_string(),
+            flags: InterfaceFlags::from_bits_truncate(info.ifa_flags as i32),
+            address: address,
+            netmask: netmask,
+            broadcast: None,
+            destination: None,
+        };
+
+        #[cfg(any(target_os = "linux", target_os = "emscripten", target_os = "fuchsia"))]
+        {
+            let ifo = unsafe { InetAddr::from_libc_sockaddr(info.ifa_ifu) };
+            if addr.flags.contains(IFF_POINTOPOINT) {
+                addr.destination = ifo;
+            } else if addr.flags.contains(IFF_BROADCAST) {
+                addr.broadcast = ifo;
+            }
+        }
+
+        #[cfg(any(target_os = "macos", target_os = "ios", target_os = "freebsd",
+                    target_os = "dragonfly", target_os = "openbsd", target_os = "netbsd",
+                    target_os = "bitrig", target_os = "solaris"))]
+        {
+            addr.destination = unsafe { InetAddr::from_libc_sockaddr(info.ifa_dstaddr) };
```

Comment:
Looks like you forgot to set the broadcast for non-Linux platforms.  You can do it the same way as on Linux.  Depending on the flags, either `ifa_dstaddr` or `ifa_broadaddr` is set.  The only difference is that on BSD, one of those is a macro for the other, whereas on Linux they're a union.

Reply:
I just checked, and libc does define both `IFF_BROADCAST` and `IFF_POINTOPOINT` for most, if not all, platforms.